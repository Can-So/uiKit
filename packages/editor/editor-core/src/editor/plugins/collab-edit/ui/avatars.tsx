import * as React from 'react';
import styled, { keyframes } from 'styled-components';
import { EditorView } from 'prosemirror-view';
import Avatar, { AvatarGroup } from '@atlaskit/avatar';
import { akGridSizeUnitless } from '@atlaskit/util-shared-styles';

import WithPluginState from '../../../ui/WithPluginState';
import { EventDispatcher } from '../../../event-dispatcher';
import { pluginKey as collabEditPluginKey, PluginState } from '../plugin';
import { getAvatarColor } from '../utils';

export interface Props {
  editorView?: EditorView;
  eventDispatcher?: EventDispatcher;
}

const AvatarContainer = styled.div`
  margin-right: ${akGridSizeUnitless}px;
`;

const itemAppear = keyframes`
0% {
  transform: scale(0);
}

50% {
  transform: scale(1.1);
}

100% {
  transform: scale(1);
}
`;

const AvatarItem: any = styled.div`
  position: relative;
  align-self: center;

  & > div {
    animation: ${itemAppear} 500ms 1;
    animation-fill-mode: both;
  }

  &::before {
    content: '${(props: any) => props.avatar}';
    display: block;
    position: absolute;
    right: -1px;
    bottom: -1px;
    width: 13px;
    height: 13px;
    z-index: 10;
    border-radius: 3px;
    background: ${(props: any) => props.badgeColor};
    color: #fff;
    font-size: 9px;
    line-height: 0;
    padding-top: 7px;
    text-align: center;
    box-shadow: 0 0 1px #fff;
    box-sizing: border-box;

    animation: ${itemAppear} 250ms 1;
    animation-fill-mode: both;
    animation-delay: 400ms;
  }
`;

declare interface ItemProps {
  name: string;
  sessionId: string;
  email: string;
  src: string;
}

function Item(props: ItemProps) {
  const color = getAvatarColor(props.sessionId).color.solid;
  const avatar = props.name.substr(0, 1).toUpperCase();

  return (
    <AvatarItem badgeColor={color} avatar={avatar}>
      <Avatar {...props} />
    </AvatarItem>
  );
}

export default class Avatars extends React.Component<Props, any> {
  private onAvatarClick = event => {};
  private renderAvatars = state => {
    if (!state.data) {
      return null;
    }
    const { sessionId, activeParticipants } = state.data as PluginState;
    const avatars = activeParticipants
      .toArray()
      .map(p => ({
        email: p.email,
        key: p.sessionId,
        name: p.name,
        src: p.avatar,
        sessionId: p.sessionId,
        size: 'medium',
      }))
      .sort(p => (p.sessionId === sessionId ? -1 : 1));

    if (!avatars.length) {
      return null;
    }

    return (
      <AvatarContainer>
        <AvatarGroup
          appearance="stack"
          size="medium"
          data={avatars}
          onAvatarClick={this.onAvatarClick}
          avatar={Item}
        />
      </AvatarContainer>
    );
  };

  render() {
    const { eventDispatcher, editorView } = this.props;

    return (
      <WithPluginState
        editorView={editorView}
        eventDispatcher={eventDispatcher}
        plugins={{ data: collabEditPluginKey }}
        render={this.renderAvatars}
      />
    );
  }
}
