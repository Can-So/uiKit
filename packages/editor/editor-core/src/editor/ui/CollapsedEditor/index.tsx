import * as React from 'react';
import Editor from '../../../editor';
import EditorWithActions from '../../../labs/EditorWithActions';
import ChromeCollapsed from '../../../ui/ChromeCollapsed';

export interface Props {
  placeholder?: string;
  children?: any;
  isExpanded?: boolean;

  onFocus?: (e) => void;
  onExpand?: () => void;
}

export interface State {}

export default class CollapsedEditor extends React.Component<Props, State> {
  editorComponent?: Editor;
  shouldTriggerExpandEvent?: boolean;

  componentWillReceiveProps(nextProps, nextState) {
    if (!this.props.isExpanded && nextProps.isExpanded) {
      this.shouldTriggerExpandEvent = true;
    }
  }

  componentDidUpdate() {
    if (this.shouldTriggerExpandEvent && this.editorComponent) {
      this.shouldTriggerExpandEvent = false;
      if (this.props.onExpand) {
        this.props.onExpand();
      }
    }
  }

  handleEditorRef = (editorRef?: Editor, editorRefCallback?: any) => {
    if (editorRefCallback && typeof editorRefCallback === 'function') {
      editorRefCallback(editorRef);
    }
    this.editorComponent = editorRef;
  };

  render() {
    const child = React.Children.only(this.props.children);
    if (child.type !== Editor && child.type !== EditorWithActions) {
      throw new Error('Expected child to be of type `Editor`');
    }

    if (!this.props.isExpanded) {
      return (
        <ChromeCollapsed
          onFocus={this.props.onFocus}
          text={this.props.placeholder}
        />
      );
    }

    return React.cloneElement(child, {
      ref: editorComponent =>
        this.handleEditorRef(editorComponent, (child as any).ref),
    });
  }
}
