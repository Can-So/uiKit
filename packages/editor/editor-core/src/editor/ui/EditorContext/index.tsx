import * as React from 'react';
import * as PropTypes from 'prop-types';
import EditorActions from '../../actions';

export default class EditorContext extends React.Component<
  { editorActions?: EditorActions },
  {}
> {
  static childContextTypes = {
    editorActions: PropTypes.object,
  };

  private editorActions: EditorActions;

  constructor(props) {
    super(props);
    this.editorActions = props.editorActions || new EditorActions();
  }

  getChildContext() {
    return {
      editorActions: this.editorActions,
    };
  }

  render() {
    return React.Children.only(this.props.children);
  }
}
