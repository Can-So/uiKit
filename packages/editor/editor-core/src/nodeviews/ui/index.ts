export interface ReactNodeProps {
  selected: boolean;
}

export type ReactComponentConstructor = new () => React.Component<any, any>;
export type ProsemirrorGetPosHandler = () => number;

export {
  default as ReactPMNode,
  ReactProsemirrorNodeProps,
} from './prosemirror-node';
