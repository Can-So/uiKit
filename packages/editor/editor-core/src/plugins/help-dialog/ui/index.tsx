import * as React from 'react';
import { Schema } from 'prosemirror-model';
import { EditorView } from 'prosemirror-view';
import { browser } from '@atlaskit/editor-common';
import CloseIcon from '@atlaskit/icon/glyph/editor/close';
import Modal from '@atlaskit/modal-dialog';
import {
  Header,
  ContentWrapper,
  Line,
  Content,
  ColumnRight,
  ColumnLeft,
  Row,
  CodeSm,
  CodeMd,
  CodeLg,
  Title,
} from './styles';
import * as keymaps from '../../../keymaps';
import ToolbarButton from '../../../ui/ToolbarButton';
import { closeHelpCommand } from '../';

// tslint:disable-next-line:variable-name
const AkModalDialog: React.ComponentClass<any> = Modal;

export interface Format {
  name: string;
  type: string;
  keymap?: Function;
  autoFormatting?: Function;
}

export const formatting: Format[] = [
  {
    name: 'Bold',
    type: 'strong',
    keymap: () => keymaps.toggleBold,
    autoFormatting: () => (
      <span>
        <CodeLg>**Bold**</CodeLg>
      </span>
    ),
  },
  {
    name: 'Italic',
    type: 'em',
    keymap: () => keymaps.toggleItalic,
    autoFormatting: () => (
      <span>
        <CodeLg>*Italic*</CodeLg>
      </span>
    ),
  },
  {
    name: 'Underline',
    type: 'underline',
    keymap: () => keymaps.toggleUnderline,
  },
  {
    name: 'Strikethrough',
    type: 'strike',
    keymap: () => keymaps.toggleStrikethrough,
    autoFormatting: () => (
      <span>
        <CodeLg>~~strikethrough~~</CodeLg>
      </span>
    ),
  },
  {
    name: 'Heading 1',
    type: 'heading',
    autoFormatting: () => (
      <span>
        <CodeSm>#</CodeSm> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Heading 5',
    type: 'heading',
    autoFormatting: () => (
      <span>
        <CodeLg>#####</CodeLg> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Numbered list',
    type: 'orderedList',
    keymap: () => keymaps.toggleOrderedList,
    autoFormatting: () => (
      <span>
        <CodeSm>1.</CodeSm> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Bulleted list',
    type: 'bulletList',
    keymap: () => keymaps.toggleBulletList,
    autoFormatting: () => (
      <span>
        <CodeSm>*</CodeSm> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Quote',
    type: 'blockquote',
    keymap: () => keymaps.toggleBlockQuote,
    autoFormatting: () => (
      <span>
        <CodeLg>></CodeLg> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Code block',
    type: 'codeBlock',
    autoFormatting: () => (
      <span>
        <CodeLg>```</CodeLg>
      </span>
    ),
  },
  {
    name: 'Divider',
    type: 'rule',
    keymap: () => keymaps.insertRule,
    autoFormatting: () => (
      <span>
        <CodeLg>---</CodeLg>
      </span>
    ),
  },
  {
    name: 'Link',
    type: 'link',
    keymap: ({ appearance }) =>
      appearance && appearance !== 'message' ? keymaps.addLink : undefined,
    autoFormatting: () => (
      <span>
        <CodeLg>[Link](http://a.com)</CodeLg>
      </span>
    ),
  },
  {
    name: 'Code',
    type: 'code',
    keymap: () => keymaps.toggleCode,
    autoFormatting: () => (
      <span>
        <CodeLg>`code`</CodeLg>
      </span>
    ),
  },
  {
    name: 'Actions',
    type: 'taskItem',
    autoFormatting: () => (
      <span>
        <CodeSm>[]</CodeSm> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Decisions',
    type: 'decisionItem',
    autoFormatting: () => (
      <span>
        <CodeSm>&lt;&gt;</CodeSm> + <CodeLg>space</CodeLg>
      </span>
    ),
  },
  {
    name: 'Emoji',
    type: 'emoji',
    autoFormatting: () => (
      <span>
        <CodeLg>:</CodeLg>
      </span>
    ),
  },
  {
    name: 'Mention',
    type: 'mention',
    autoFormatting: () => (
      <span>
        <CodeLg>@</CodeLg>
      </span>
    ),
  },
];

export const getSupportedFormatting = (schema: Schema): Format[] => {
  return formatting.filter(
    format => schema.nodes[format.type] || schema.marks[format.type],
  );
};

export const getComponentFromKeymap = (keymap): any => {
  const currentMap = keymap[browser.mac ? 'mac' : 'windows'];
  const keyParts = currentMap.replace(/\-(?=.)/g, ' + ').split(' ');
  return (
    <span>
      {keyParts.map((part, index) => {
        if (part === '+') {
          return <span key={`${currentMap}-${index}`}>{' + '}</span>;
        } else if (part === 'Cmd') {
          return <CodeSm key={`${currentMap}-${index}`}>âŒ˜</CodeSm>;
        } else if (
          ['ctrl', 'alt', 'opt', 'shift'].indexOf(part.toLowerCase()) >= 0
        ) {
          return (
            <CodeMd key={`${currentMap}-${index}`}>{part.toLowerCase()}</CodeMd>
          );
        }
        return <CodeSm key={`${currentMap}-${index}`}>{part}</CodeSm>;
      })}
    </span>
  );
};

export interface Props {
  editorView: EditorView;
  isVisible: boolean;
  appearance?: string;
}

// tslint:disable-next-line:variable-name
const ModalHeader = ({ onClose, showKeyline }) => (
  <Header showKeyline={showKeyline}>
    Keyboard shortcuts
    <div>
      <ToolbarButton
        onClick={onClose}
        title="Close help dialog"
        iconBefore={<CloseIcon label="Close help dialog" size="large" />}
      />
    </div>
  </Header>
);

export default class HelpDialog extends React.Component<Props, any> {
  private formatting: Format[];

  constructor(props) {
    super(props);
    const { schema } = this.props.editorView.state;
    this.formatting = getSupportedFormatting(schema);
  }

  closeDialog = () => {
    const { state: { tr }, dispatch } = this.props.editorView;
    closeHelpCommand(tr, dispatch);
  };

  handleEsc = e => {
    if (e.key === 'Escape' && this.props.isVisible) {
      this.closeDialog();
    }
  };

  componentDidMount() {
    document.addEventListener('keydown', this.handleEsc);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleEsc);
  }

  render() {
    if (!this.props.isVisible) {
      return null;
    }

    return (
      <AkModalDialog
        width="large"
        onClose={this.closeDialog}
        header={ModalHeader}
      >
        <ContentWrapper>
          <Line />
          <Content>
            <ColumnLeft>
              <Title>Text Formatting</Title>
              <div>
                {this.formatting
                  .filter(form => {
                    const keymap = form.keymap && form.keymap(this.props);
                    return keymap && keymap[browser.mac ? 'mac' : 'windows'];
                  })
                  .map(form => (
                    <Row key={`textFormatting-${form.name}`}>
                      <span>{form.name}</span>
                      {getComponentFromKeymap(
                        form.keymap!({ appearance: this.props.appearance }),
                      )}
                    </Row>
                  ))}
              </div>
            </ColumnLeft>
            <ColumnRight>
              <Title>Markdown</Title>
              <div>
                {this.formatting.map(
                  form =>
                    form.autoFormatting && (
                      <Row key={`autoFormatting-${form.name}`}>
                        <span>{form.name}</span>
                        {form.autoFormatting()}
                      </Row>
                    ),
                )}
              </div>
            </ColumnRight>
          </Content>
        </ContentWrapper>
      </AkModalDialog>
    );
  }
}
