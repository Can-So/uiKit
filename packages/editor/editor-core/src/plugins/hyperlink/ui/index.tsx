import * as React from 'react';
import { Node, Mark } from 'prosemirror-model';
import { EditorView } from 'prosemirror-view';
import {
  setLinkText,
  removeLink,
  hideLinkToolbar,
  setLinkHref,
  insertLink,
} from '../commands';
import HyperlinkEdit from './HyperlinkEdit';
import { findDomRefAtPos } from 'prosemirror-utils';
import { ActivityProvider } from '@atlaskit/activity';
import RecentSearch from './RecentSearch';
import { normalizeUrl } from '../utils';
import { InsertStatus, HyperlinkState } from '../pm-plugins/main';

export class AddDisplayTextToolbar extends React.PureComponent<{
  pos: number;
  node: Node;
  view: EditorView;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
}> {
  render() {
    const {
      pos,
      node,
      view,
      popupsMountPoint,
      popupsBoundariesElement,
    } = this.props;
    const existingLink = (node.type.schema.marks.link.isInSet(
      node.marks,
    ) as Mark).attrs.href;
    const unlink = () =>
      removeLink(pos)(view.state, view.dispatch) && view.focus();
    const hideToolbar = () =>
      hideLinkToolbar()(view.state, view.dispatch) && view.focus();
    const updateLinkText = (text: string) =>
      setLinkText(pos, text)(view.state, view.dispatch) && view.focus();
    const updateLinkTextOrHideToolbar = (text: string) =>
      updateLinkText(text) || hideToolbar();
    return (
      <HyperlinkEdit
        target={findDomRefAtPos(pos, view.domAtPos.bind(view))}
        popupsMountPoint={popupsMountPoint}
        popupsBoundariesElement={popupsBoundariesElement}
        alwaysOpenLinkAt={existingLink}
        placeholder="Text to display"
        onSubmit={updateLinkText}
        onBlur={updateLinkTextOrHideToolbar}
        onUnlink={unlink}
        onOpenLink={() => {}}
      />
    );
  }
}

export class EditLinkHrefToolbar extends React.PureComponent<{
  pos: number;
  node: Node;
  view: EditorView;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
}> {
  render() {
    const {
      pos,
      node,
      view,
      popupsMountPoint,
      popupsBoundariesElement,
    } = this.props;
    const existingLink = (node.type.schema.marks.link.isInSet(
      node.marks,
    ) as Mark).attrs.href;
    const hideToolbar = () =>
      hideLinkToolbar()(view.state, view.dispatch) && view.focus();
    const updateLinkHref = (href: string) =>
      setLinkHref(pos, href)(view.state, view.dispatch) && view.focus();
    const updateLinkHrefOrHideToolbar = (href: string) =>
      updateLinkHref(href) || hideToolbar();
    const unlink = () =>
      removeLink(pos)(view.state, view.dispatch) && view.focus();
    return (
      <HyperlinkEdit
        target={findDomRefAtPos(pos, view.domAtPos.bind(view))}
        popupsMountPoint={popupsMountPoint}
        popupsBoundariesElement={popupsBoundariesElement}
        defaultValue={existingLink}
        placeholder="Paste link"
        onSubmit={updateLinkHref}
        onBlur={updateLinkHrefOrHideToolbar}
        onUnlink={unlink}
        onOpenLink={() => {}}
      />
    );
  }
}

export class InsertLinkToolbar extends React.PureComponent<{
  from: number;
  to: number;
  view: EditorView;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
}> {
  render() {
    const {
      from,
      to,
      view,
      popupsMountPoint,
      popupsBoundariesElement,
    } = this.props;
    const hideToolbar = () =>
      hideLinkToolbar()(view.state, view.dispatch) && view.focus();
    const addLink = (href: string) =>
      insertLink(from, to, href)(view.state, view.dispatch) && view.focus();
    return (
      <HyperlinkEdit
        target={findDomRefAtPos(from, view.domAtPos.bind(view))}
        popupsMountPoint={popupsMountPoint}
        popupsBoundariesElement={popupsBoundariesElement}
        autoFocus={true}
        placeholder="Paste link"
        onSubmit={addLink}
        onBlur={hideToolbar}
      />
    );
  }
}

export class ActivityPoweredInsertLinkToolbar extends React.PureComponent<{
  from: number;
  to: number;
  view: EditorView;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
  activityProvider: Promise<ActivityProvider>;
}> {
  render() {
    const {
      from,
      to,
      view,
      popupsMountPoint,
      popupsBoundariesElement,
      activityProvider,
    } = this.props;
    const hideToolbar = () =>
      hideLinkToolbar()(view.state, view.dispatch) && view.focus();
    const addLink = (href: string, text?: string) =>
      insertLink(from, to, href, text)(view.state, view.dispatch) &&
      view.focus();
    return (
      <RecentSearch
        target={findDomRefAtPos(from, view.domAtPos.bind(view))}
        popupsMountPoint={popupsMountPoint}
        popupsBoundariesElement={popupsBoundariesElement}
        autoFocus={true}
        activityProvider={activityProvider}
        placeholder="Paste link or search recently viewed"
        onSubmit={addLink}
        onBlur={hideToolbar}
      />
    );
  }
}

export interface Props {
  hyperlinkState?: HyperlinkState;
  view: EditorView;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
  activityProvider?: Promise<ActivityProvider>;
}
export default function HyperlinkToolbar(props: Props) {
  const {
    hyperlinkState,
    view,
    popupsMountPoint,
    popupsBoundariesElement,
    activityProvider,
  } = props;
  if (hyperlinkState && hyperlinkState.activeLinkMark) {
    if (hyperlinkState.activeLinkMark.type === InsertStatus.EDIT_LINK_TOOLBAR) {
      const { node, pos } = hyperlinkState.activeLinkMark;
      const mark = view.state.schema.marks.link.isInSet(node.marks) as Mark;
      const isLinkTextTheSameAsTheLinkUrl =
        mark.attrs.href === normalizeUrl(node.text!);
      const Toolbar = isLinkTextTheSameAsTheLinkUrl
        ? AddDisplayTextToolbar
        : EditLinkHrefToolbar;
      return (
        <Toolbar
          pos={pos}
          node={node}
          view={view}
          popupsMountPoint={popupsMountPoint}
          popupsBoundariesElement={popupsBoundariesElement}
        />
      );
    } else if (
      hyperlinkState.activeLinkMark.type === InsertStatus.INSERT_LINK_TOOLBAR
    ) {
      const { from, to } = hyperlinkState.activeLinkMark;
      if (activityProvider) {
        return (
          <ActivityPoweredInsertLinkToolbar
            from={from}
            to={to}
            view={view}
            popupsMountPoint={popupsMountPoint}
            popupsBoundariesElement={popupsBoundariesElement}
            activityProvider={activityProvider}
          />
        );
      } else {
        return (
          <InsertLinkToolbar
            from={from}
            to={to}
            view={view}
            popupsMountPoint={popupsMountPoint}
            popupsBoundariesElement={popupsBoundariesElement}
          />
        );
      }
    }
  }
  return null;
}
