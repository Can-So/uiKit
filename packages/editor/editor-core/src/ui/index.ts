export { default as EmojiTypeAhead } from './EmojiTypeAhead';
export { default as HyperlinkEdit } from './HyperlinkEdit';
export { default as LanguagePicker } from './LanguagePicker';
export { DEFAULT_LANGUAGES } from './LanguagePicker/languageList';
export { default as MentionPicker } from './MentionPicker';
export { default as PanelTextInput } from './PanelTextInput';
export { default as ToolbarBlockType } from './ToolbarBlockType';
export { default as ToolbarButton } from './ToolbarButton';
export { default as ToolbarDecision } from './ToolbarDecision';
export { default as ToolbarEmojiPicker } from './ToolbarEmojiPicker';
export { default as ToolbarFeedback } from './ToolbarFeedback';
export { default as ToolbarLists } from './ToolbarLists';
export { default as ToolbarTask } from './ToolbarTask';
export { default as ToolbarTextFormatting } from './ToolbarTextFormatting';
export { default as ToolbarInsertBlock } from './ToolbarInsertBlock';
export { default as ToolbarMedia } from './ToolbarMedia';
export { default as UnsupportedBlock } from './UnsupportedBlock';
