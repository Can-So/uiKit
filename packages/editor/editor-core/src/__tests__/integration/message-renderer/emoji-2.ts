import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import { getDocFromElement } from '../_helpers';
import {
  messageEditor,
  editable,
  insertEmoji,
  emojiItem,
  insertEmojiBySelect,
} from './_emoji-helpers';
import { messages } from '../../../plugins/insert-block/ui/ToolbarInsertBlock';

BrowserTestCase(
  'Emoji: should be able to use emoji inside blockquote',
  { skip: ['ie'] },
  async client => {
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.type(editable, '> ');
    await browser.type(editable, 'some text ');
    await insertEmoji(browser, 'a');
    await browser.waitForSelector(emojiItem('a'));
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Emoji: should be able to use emoji inside bulletList',
  { skip: ['ie'] },
  async client => {
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.type(editable, '* ');
    await insertEmoji(browser, 'smile');
    await browser.waitForSelector(emojiItem('smile'));
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Emoji: should be able to use emoji inside orderedList',
  { skip: ['ie'] },
  async client => {
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.type(editable, '1. ');
    await insertEmoji(browser, 'a');
    await browser.waitForSelector(emojiItem('a'));
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

// ie keying in ; instead of : - browserstack issue
BrowserTestCase(
  'Emoji: should be able remove emoji on backspace',
  { skip: ['safari', 'ie'] },
  async client => {
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.waitForSelector(editable);
    await browser.type(editable, 'this ');
    await insertEmoji(browser, 'joy');
    await browser.waitForSelector(emojiItem('joy'));
    await browser.type(editable, ['Backspace', 'Backspace']);
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

// Safari does not understand webdriver click
// IE has a bug opening picker inside task/decisions
BrowserTestCase(
  'Emoji: should be able to select emoji by clicking inside decisions',
  { skip: ['safari', 'ie'] },
  async client => {
    const decisions = 'span[aria-label="Decision"]';
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.waitForSelector(editable);
    // to get steps working on edge since its is slow
    await browser.type(editable, '<> ');
    await browser.waitForSelector(decisions);
    await browser.type(editable, 'this ');
    await insertEmojiBySelect(browser, 'smile');
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'Emoji: should be able to change text with emoji into decisions',
  { skip: ['ie'] },
  async client => {
    const decisions = 'li span[aria-label="Decision"]';
    const createDecisions = `span[aria-label="${
      messages.decision.defaultMessage
    }"]`;
    const browser = await new Page(client);
    await browser.goto(messageEditor);
    await browser.waitForSelector(editable);
    await browser.type(editable, 'this ');
    await insertEmoji(browser, 'smile');
    await browser.waitForSelector(createDecisions);
    await browser.click(createDecisions);
    await browser.waitForSelector(decisions);
    await browser.isExisting(decisions);
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
