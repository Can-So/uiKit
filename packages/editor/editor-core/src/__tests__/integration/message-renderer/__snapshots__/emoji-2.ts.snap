// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emoji-2.ts: should be able remove emoji on backspace 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "this ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`emoji-2.ts: should be able to change text with emoji into decisions 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "attrs": Object {
            "localId": "abc-123",
            "state": "DECIDED",
          },
          "content": Array [
            Object {
              "text": "this ",
              "type": "text",
            },
            Object {
              "attrs": Object {
                "id": "1f604",
                "shortName": ":smile:",
                "text": "ðŸ˜„",
              },
              "type": "emoji",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "decisionItem",
        },
      ],
      "type": "decisionList",
    },
  ],
  "type": "doc",
}
`;

exports[`emoji-2.ts: should be able to select emoji by clicking inside decisions 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "attrs": Object {
            "localId": "abc-123",
            "state": "DECIDED",
          },
          "content": Array [
            Object {
              "text": "this ",
              "type": "text",
            },
            Object {
              "attrs": Object {
                "id": "1f604",
                "shortName": ":smile:",
                "text": "ðŸ˜„",
              },
              "type": "emoji",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "decisionItem",
        },
      ],
      "type": "decisionList",
    },
  ],
  "type": "doc",
}
`;

exports[`emoji-2.ts: should be able to use emoji inside blockquote 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "some text ",
              "type": "text",
            },
            Object {
              "attrs": Object {
                "id": "1f170",
                "shortName": ":a:",
                "text": "ðŸ…°",
              },
              "type": "emoji",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "blockquote",
    },
  ],
  "type": "doc",
}
`;

exports[`emoji-2.ts: should be able to use emoji inside bulletList 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "attrs": Object {
                    "id": "1f604",
                    "shortName": ":smile:",
                    "text": "ðŸ˜„",
                  },
                  "type": "emoji",
                },
                Object {
                  "text": " ",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`emoji-2.ts: should be able to use emoji inside orderedList 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "attrs": Object {
                    "id": "1f170",
                    "shortName": ":a:",
                    "text": "ðŸ…°",
                  },
                  "type": "emoji",
                },
                Object {
                  "text": " ",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "orderedList",
    },
  ],
  "type": "doc",
}
`;
